binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/properties/KafkaStreamsConsumerProperties.java

```diff
@@ -71,6 +71,12 @@ public class KafkaStreamsConsumerProperties extends KafkaConsumerProperties {
 	 */
 	private String consumedAs;
 
+	/**
+	 * When event type based routing is enabled, the binder uses the byte[] Serde by default.
+	 * Use this property to override this default behavior by forcing the binder to use the configured or inferred Serde.
+	 */
+	private boolean useConfiguredSerdeWhenRoutingEvents;
```

Comment:
**[NIT]** The name is a bit long, but I don't really have a shorter better alternate recommendation.

Reply:
Ya, let me know if you have a better alternative. :-)

Reply:
If there was an `event-routing` sub-properties that contained `supported-types` and `header-key` it would fit in nicely as `use-configured-serde`. However, I think w/ the flat nature of the props currently this is fine.